### PR TITLE
Add push notifications backend with parse provider

### PR DESCRIPTION
**Pull request by joiggama**
Depends on #3

This only adds the backend structure for push notifications without any implemented.

### CODE REVIEW

File: spree_sam.gemspec [hunk 1]
```diff
@@ -16,7 +16,8 @@ Gem::Specification.new do |s|
   s.require_path = 'lib'
   s.requirements << 'none'
 
-  s.add_dependency 'spree_core', '~> 3.1.0.beta'
+  s.add_dependency 'httparty', '~> 0.13.7'
+  s.add_dependency 'spree_core', '~> 3.0.5.beta'
```
Comment (joiggama) [inline spree_sam.gemspec:20]: Please, ignore this, it comes from #3.

File: lib/spree_sam/notifications/providers/parse.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require "httparty"
+
+module SpreeSam
+  module Notifications
+    module Providers
+      class Parse
```
Comment (softr8) [inline lib/spree_sam/notifications/providers/parse.rb]: I really liked this!

quick note:
if you are going to build the possibility of specifying different providers, let's make a parent class with all the shared methods and the ones that have to be implemented, for example: a method build_payload, which 'decorate' the hash payload to customize params required by the strategy

Comment (joiggama) [inline lib/spree_sam/notifications/providers/parse.rb]: Makes sense, thanks, I'll do that.

**Comment by softr8** [comment]
Nice!

**Pull request merged**